### ISSUE TITLE
When I wrote a plugin for use with logger++, some unintended errors occurred.

### ISSUE THREAD

**Issue by 17307** [nccgroup/LoggerPlusPlus#190]
**Description:**

When I wrote a plugin for use with logger++, some unintended errors occurred.
I'm using the burp.api.montoya.MontoyaApi version of the API, and when I condition a notes for annotations in handleRequestReceived, I find that my notes are overwritten by logger++. can logger++ not directly overwrite the notes, but instead get a bit of information about the notes from other plugins?

Here is the code

```
return ProxyRequestReceivedAction.continueWith(httpRequestNew, annotations.withNotes("request1"));
```

The `request1` message I added gets overwritten by the logger++ one.

**Steps To Reproduce:**

**Expected behavior:**

**Screenshots:**

**Version:**

**Comment by C0DEbrained** [comment]
Unfortunately, the Burp API doesn't provide a way for two plugins to write separate comments to a single HTTP request.

Though, to maintain compatibility, Logger++ should append it's comment to any existing comment set by other tools, then remove the tag once the response has been processed, leaving the original comment intact. As long as L++ is the last plugin in the list, this should be invisible to other apps using the annotations functionality.

**Comment by 17307** [comment]
> Unfortunately, the Burp API doesn't provide a way for two plugins to write separate comments to a single HTTP request.
> 
> Though, to maintain compatibility, Logger++ should append it's comment to any existing comment set by other tools, then remove the tag once the response has been processed, leaving the original comment intact. As long as L++ is the last plugin in the list, this should be invisible to other apps using the annotations functionality.

Yes, in the current version even if I put logger++ at the end of the plugin list, my note is still overwritten.
